core/src/main/scala/kafka/coordinator/transaction/TransactionMetadata.scala

```diff
@@ -25,8 +25,50 @@ import org.apache.kafka.common.record.RecordBatch
 
 import scala.collection.{immutable, mutable}
 
+
+object TransactionState {
+  val AllStates = Set(
+    Empty,
+    Ongoing,
+    PrepareCommit,
+    PrepareAbort,
+    CompleteCommit,
+    CompleteAbort,
+    Dead,
+    PrepareEpochFence
+  )
+
+  def fromName(name: String): Option[TransactionState] = {
+    name match {
+      case "Empty" => Some(Empty)
+      case "Ongoing" => Some(Ongoing)
+      case "PrepareCommit" => Some(PrepareCommit)
+      case "PrepareAbort" => Some(PrepareAbort)
+      case "CompleteCommit" => Some(CompleteCommit)
+      case "CompleteAbort" => Some(CompleteAbort)
+      case "PrepareEpochFence" => Some(PrepareEpochFence)
+      case "Dead" => Some(Dead)
+      case _ => None
+    }
+  }
+
+  def fromId(id: Byte): TransactionState = {
```

Comment:
How about 'AllStates.find(_.id == id)'?

Reply:
I left this one as it was since I am not 100% sure whether it has any impact on the performance of transaction loading.